### DISCUSSION TITLE
Unable to specify a non-latest jetstream version when configuring an EventBus

### DISCUSSION THREAD

**Discussion by smgtx** [argoproj/argo-events#2313] [Q&A]
As the title states, I am unable to specify a non-latest jetstream version when configuring an EventBus.

Tested with argo-events 1.7.3, and deployed using argo-events helm chart 2.0.6.

I am getting the following errors from the argo-events controller:
`{"level":"error","ts":1668552858.8801756,"logger":"argo-events.eventbus-controller","caller":"eventbus/controller.go:56","msg":"reconcile error","namespace":"argo","eventbus":"default","error":"failed to get jetstream version, err: unsupported version \"2.8.1\", supported versions: \"latest\"","stacktrace":"github.com/argoproj/argo-events/controllers/eventbus.(*reconciler).Reconcile\n\t/home/runner/work/argo-events/argo-events/controllers/eventbus/controller.go:56\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.2/pkg/internal/controller/controller.go:114\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.2/pkg/internal/controller/controller.go:311\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.2/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.2/pkg/internal/controller/controller.go:227"}`

and from the EventBus

```apiVersion: argoproj.io/v1alpha1
kind: EventBus
metadata:
  name: default
spec:
  jetstream:
    version: 2.8.1
```

`failed to get jetstream version, err: unsupported version "2.8.1", supported versions: "latest" `

This error makes sense in the context of the generated yaml:
```# Source: argo-events/templates/argo-events-controller/config.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: argo-events-controller-manager
  labels:
    helm.sh/chart: argo-events-2.0.6
    app.kubernetes.io/name: argo-events-controller-manager
    app.kubernetes.io/instance: argo-events
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/part-of: argo-events
data:
  controller-config.yaml: |
    eventBus:
      nats:
        versions:
        - version: latest
          natsStreamingImage: nats-streaming:latest
          metricsExporterImage: natsio/prometheus-nats-exporter:latest
      jetstream:
        # Default JetStream settings, could be overridden by EventBus JetStream specs
        settings: |
          # https://docs.nats.io/running-a-nats-service/configuration#jetstream
          # Only configure "max_memory_store" or "max_file_store", do not set "store_dir" as it has been hardcoded.
          max_memory_store: -1
          max_file_store: -1
        # The default properties of the streams to be created in this JetStream service
        streamConfig: |
          maxMsgs: 1e+06
          maxAge: 72h
          maxBytes: 1GB
          replicas: 3
          duplicates: 300s
        versions:
        - version: latest
          natsImage: nats:latest
          metricsExporterImage: natsio/prometheus-nats-exporter:latest
          configReloaderImage: natsio/nats-server-config-reloader:latest
          startCommand: /nats-server
```

I expected there to be more supported versions in the controller-config.yaml section.

For reference, these are the values I am specifying for my helm chart:
```global:
        image: 
          tag: v1.7.3
          repository: {internal-registry}/mirror/quay.io/argoproj/argo-events
      webhook:
        serviceAccount:
          name: argo-events-webhook-sa
      controller:
        serviceAccount:
          name: argo-events-sa
      config:
        jetstream:
          versions:
            - version: "2.8.1"
              configReloaderImage: "{internal-registry}/mirror/natsio/nats-server-config-reloader:0.7.4"
              metricsExporterImage: "{internal-registry}/mirror/natsio/prometheus-nats-exporter:0.10.1"
              natsImage: "{internal-registry}/mirror/nats:2.8.1"
              startCommand: "/nats-server"
```

I would appreciate any guidance on working through this issue.

**Comment by antoinetran** [comment]
If you use helm, you can override the versions and provide your own in https://github.com/argoproj/argo-helm/blob/main/charts/argo-events/values.yaml#L70  . I tried that, it works.
